modules/rss.py

```diff
@@ -37,14 +31,22 @@ def _format_entry(self, server, channel, feed_title, entry, shorten):
                 pass
         link = "%s" % link if link else ""
 
-        feed_title_str = "%s" % feed_title if feed_title else ""
+        variables = dict(
+            longtitle=feed_title or "",
+            title=utils.parse.line_normalise(utils.http.strip_html(
+                entry["title"])),
+            link=link or "",
+            author=entry.get("author", "unknown author") or "",
+        )
+
         # just in case the format starts keyerroring and you're not sure why
         self.log.trace("RSS Entry: " + str(entry))
         try:
-            format = channel.get_setting("rss-format", "$longtitle: $title by $author - $link").replace("$longtitle", feed_title_str).replace("$title", title).replace("$link", link).replace("$author", author).format(**entry)
+            template = channel.get_setting("rss-format", "$longtitle: $title by $author - $link")
+            format = format_token_replace(template, variables).format(**entry)
```

Comment:
`format_token_replace()` returns a tuple but we only care about the formatted string bit here. we also no longer need the `.format()` because we've merged the `entry` dict with the `variables` dict.
```suggestion
            _, format = utils.parse.format_token_replace(template, variables)
```